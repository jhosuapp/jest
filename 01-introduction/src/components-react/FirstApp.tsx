
interface IFirstAppProps {
    title: string;
    subtitle: string;
    name: string;
}

const FirstApp = ({ title, subtitle, name }:IFirstAppProps ):JSX.Element => {
    return (
        <>
            <h1>{ title }</h1>
            <h2>{ subtitle }</h2>
            <p>{ name }</p>
        </>
    )
}



export { FirstApp }