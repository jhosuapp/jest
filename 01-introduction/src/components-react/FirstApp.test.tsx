import { render } from "@testing-library/react";
import { FirstApp } from "./FirstApp";

describe('Testing in FirstApp.tsx', ()=>{

    const defaultProps = {
        title: "Titulo",
        subtitle: "Subtitulo",
        name: "Nombre"
    }

    it('Should render component', ()=>{
        const { container } = render(<FirstApp { ...defaultProps } />)
        expect(container).toMatchSnapshot();
    });
    
    it('Should render h1', ()=>{
        const { container, getByText } = render(<FirstApp { ...defaultProps } />)
        expect( getByText(defaultProps.title) ).toBeTruthy();
    
        const h1 = container.querySelector('h1');
    
        expect(h1?.innerHTML).toContain( defaultProps.title );
    });


    it('Should render h2', ()=>{
        const { getByTestId } = render(<FirstApp {...defaultProps} />)

        expect(getByTestId('test-subtitle').innerHTML).toContain(defaultProps.subtitle);
    });

});
