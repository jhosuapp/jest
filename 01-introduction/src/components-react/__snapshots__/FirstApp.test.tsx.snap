// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing in FirstApp.tsx Should render component 1`] = `
<div>
  <h1>
    Titulo
  </h1>
  <h2
    data-testid="test-subtitle"
  >
    Subtitulo
  </h2>
  <p>
    Nombre
  </p>
</div>
`;
